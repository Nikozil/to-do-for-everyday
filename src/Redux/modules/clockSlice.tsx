import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { getTime } from 'date-fns';
import { AppThunk } from '../store';

export const initialState = {
  time: getTime(new Date()) as number,
  interval: undefined as number | undefined,
};

export type initialStateType = typeof initialState;
const clockSlice = createSlice({
  name: 'clock',
  initialState,
  reducers: {
    TICK: (state) => {
      return {
        ...state,
        time: getTime(new Date()),
      };
    },
    setClockInterval: (state, action: PayloadAction<any>) => {
      return {
        ...state,
        interval: action.payload,
      };
    },
  },
});

export const { TICK, setClockInterval } = clockSlice.actions;

export const startClock = (): AppThunk => (dispatch, getState) => {
  dispatch(stopClock());

  const interval = setInterval(() => {
    dispatch(TICK());
  }, 1000);

  dispatch(setClockInterval(interval));
};
export const stopClock = (): AppThunk => (dispatch, getState) => {
  clearInterval(getState().clock.interval);
  dispatch(setClockInterval(null));
};

export default clockSlice;
