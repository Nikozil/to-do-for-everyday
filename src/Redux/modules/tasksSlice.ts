import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import {
  add,
  endOfToday,
  format,
  getTime,
  milliseconds,
  startOfDay,
} from 'date-fns';

import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';

export const initialState = {
  tasksList: [] as Task[],
  livedDay: {
    score: 0,
    tag: '',
    doneTasksList: [],
    timestamp: null,
  } as LivedDay,
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const tasksSlice = createSlice({
  name: 'tasks',
  initialState,
  reducers: {
    setTasks: (state, action: PayloadAction<Task[]>) => {
      return { ...state, tasksList: action.payload };
    },
    appendNewTask: (state, action: PayloadAction<Task>) => {
      state.tasksList.push(action.payload);
    },
    editTask: (state, action: PayloadAction<PartialTask>) => {
      const task = state.tasksList.find(
        (task) => task.id === action.payload.id
      );
      if (task) {
        task.data = { ...task.data, ...action.payload.data };
      }
    },
    removeTask: (state, action: PayloadAction<string>) => {
      state.tasksList = state.tasksList.filter(
        (task) => task.id !== action.payload
      );
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
    setLivedDay: (state, action: PayloadAction<LivedDay>) => {
      state.livedDay = { ...state.livedDay, ...action.payload };
    },
    appendTaskToDoneTasksList: (state, action: PayloadAction<LivedTask>) => {
      state.livedDay.doneTasksList.push(action.payload);
    },
    removeTaskToDoneTasksList: (state, action: PayloadAction<string>) => {
      const newdoneTasksList = state.livedDay.doneTasksList.filter(
        (task) => task.id !== action.payload
      );
      state.livedDay.doneTasksList = newdoneTasksList;
    },
    setTag: (state, action: PayloadAction<string>) => {
      state.livedDay.tag = action.payload;
    },
    setScore: (state, action: PayloadAction<Score>) => {
      state.livedDay.score = action.payload;
    },
    setTimestamp: (state, action: PayloadAction<number>) => {
      state.livedDay.timestamp = action.payload;
    },
  },
});

export const {
  setTasks,
  appendNewTask,
  editTask,
  removeTask,
  setInitStatus,
  setLivedDay,
  appendTaskToDoneTasksList,
  removeTaskToDoneTasksList,
  setTag,
  setScore,
  setTimestamp,
} = tasksSlice.actions;

export const getTasks = (): AppThunk => async (dispatch, getState) => {
  try {
    let tasks = await StoreAPI.getTask();
    let time = getState().clock.time;

    let livedDay = await StoreAPI.getLivedDay(format(time, 'dd.MM.yyyy'));

    if (!tasks) throw new Error('Не удалось получить данные');
    if (!livedDay) throw new Error('Не удалось получить данные');

    dispatch(setTasks(tasks));
    dispatch(setLivedDay(livedDay));
    dispatch(setInitStatus(true));
  } catch (err: any) {
    return err.message as string;
  }
};
export const addTask =
  (taskName: string, duration: Duration, repeat = 0): AppThunk =>
  async (dispatch, getState) => {
    const time = getTime(add(getState().clock.time, duration));
    const taskData = { name: taskName, done: false, time, repeat } as TaskData;
    try {
      const newTask = await StoreAPI.setTask(taskData);
      if (!newTask) throw new Error('Не удалось создать задачу');
      dispatch(appendNewTask(newTask));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const updateTask =
  (id: string, data: PartialTaskData): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.updateTask(id, data);
      dispatch(editTask({ id, data }));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { name, done, time, repeat } = data;
      let newData;
      // make the data repeat date or complete
      if (repeat) {
        //move task for future
        let newTime = time;
        //count expiration of task + 1 repeat
        let expiration = Math.round(
          (getTime(endOfToday()) - newTime) /
            (milliseconds({ days: 1 }) * repeat) +
            1
        );
        //count date of future task
        newTime = getTime(add(newTime, { days: repeat * expiration }));
        newData = { time: newTime };
      } else {
        newData = { done: !done };
      }
      // add task to done list
      dispatch(checkupTimestamp());
      const doneTasks = {
        id: id,
        name: name,
      };
      //send data as batch
      const taskApiData = { taskId: id, taskData: newData };
      const doneTaskApiData = {
        doneTaskDate: format(getState().clock.time, 'dd.MM.yyyy'),
        doneTask: doneTasks,
        doneTaskRemove: false,
      };
      await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);
      //update store
      dispatch(editTask({ id, data: newData }));
      dispatch(appendTaskToDoneTasksList(doneTasks));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const uncheckTask =
  (task: LivedTask): AppThunk =>
  async (dispatch, getState) => {
    const { id, name } = task;
    try {
      let task = getState().tasks.tasksList.find((task) => task.id === id);
      if (task) {
        const { done, time, repeat } = task.data;
        let newData;
        // make the data repeat date or complete
        if (repeat) {
          //move date to past
          newData = { time: getTime(add(time, { days: -repeat })) };
        } else {
          //return task time and status
          newData = done
            ? { done: !done }
            : { done: false, time: getState().clock.time };
        }
        const taskApiData = { taskId: id, taskData: newData };
        const doneTask = {
          id: id,
          name: name,
        };
        const doneTaskApiData = {
          doneTaskDate: format(getState().clock.time, 'dd.MM.yyyy'),
          doneTask: doneTask,
          doneTaskRemove: true,
        };
        await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);

        //update store

        dispatch(editTask({ id, data: newData }));
        dispatch(removeTaskToDoneTasksList(id));
      } else throw new Error('Нет такой задачи');
    } catch (err: any) {
      return err.message as string;
    }
  };
export const deleteTask =
  (id: string): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.deleteTask(id);
      dispatch(removeTask(id));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const addTag =
  (tag: string): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(checkupTimestamp());
      const time = format(getState().clock.time, 'dd.MM.yyyy');
      const livedDay = { tag };
      await StoreAPI.updateLivedDay(time, livedDay);
      dispatch(setTag(tag));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const addScore =
  (score: Score): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(checkupTimestamp());
      const date = format(getState().clock.time, 'dd.MM.yyyy');
      const livedDay = { score };
      await StoreAPI.updateLivedDay(date, livedDay);
      dispatch(setScore(score));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkupTimestamp = (): AppThunk => async (dispatch, getState) => {
  if (!getState().tasks.livedDay.timestamp) {
    try {
      const date = format(getState().clock.time, 'dd.MM.yyyy');

      const timestamp = getTime(startOfDay(getState().clock.time));
      const livedDay = { timestamp };
      await StoreAPI.updateLivedDay(date, livedDay);
      dispatch(setTimestamp(timestamp));
    } catch (err: any) {
      return err.message as string;
    }
  }
};

export const doAgainTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { done } = data;

      //set date of tomorrow task
      const currentTime = getState().clock.time;
      const newTime = getTime(add(currentTime, { days: 1 }));

      const newData = { time: newTime, done: !done };

      await StoreAPI.updateTask(id, newData);
      //update store
      dispatch(editTask({ id, data: newData }));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default tasksSlice;

export interface Task {
  id: string;
  data: TaskData;
}
export interface TaskData {
  name: string;
  done: boolean;
  time: number;
  repeat: number;
}
export type PartialTaskData = Partial<TaskData>;
export interface PartialTask {
  id: string;
  data: PartialTaskData;
}
export interface LivedTask {
  id: string;
  name: string;
}
export interface LivedDay {
  tag: string;
  score: Score;
  doneTasksList: LivedTask[];
  timestamp: null | number;
}
export type Score = 0 | 1 | 2 | 3 | 4 | 5;
