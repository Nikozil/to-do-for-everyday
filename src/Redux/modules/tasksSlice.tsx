import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { add, endOfToday, format, getTime, milliseconds } from 'date-fns';
import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';

export const initialState = {
  tasksList: [] as Task[],
  doneTasksList: {} as DoneTask,
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const tasksSlice = createSlice({
  name: 'tasks',
  initialState,
  reducers: {
    setTasks: (state, action: PayloadAction<Task[]>) => {
      return { ...state, tasksList: action.payload };
    },
    editTask: (state, action: PayloadAction<PartialTask>) => {
      const task = state.tasksList.find(
        (task) => task.id === action.payload.id
      );
      if (task) {
        task.data = { ...task.data, ...action.payload.data };
      }
    },
    removeTask: (state, action: PayloadAction<string>) => {
      state.tasksList = state.tasksList.filter(
        (task) => task.id !== action.payload
      );
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
    setDoneTasks: (state, action: PayloadAction<DoneTask>) => {
      return { ...state, doneTasksList: action.payload };
    },
    addTaskToDoneTasksList: (state, action: PayloadAction<DoneTask>) => {
      return {
        ...state,
        doneTasksList: { ...state.doneTasksList, ...action.payload },
      };
    },
    removeTaskToDoneTasksList: (state, action: PayloadAction<string>) => {
      delete state.doneTasksList[action.payload];
    },
  },
});

export const {
  setTasks,
  editTask,
  removeTask,
  setInitStatus,
  setDoneTasks,
  addTaskToDoneTasksList,
  removeTaskToDoneTasksList,
} = tasksSlice.actions;

export const getTasks = (): AppThunk => async (dispatch) => {
  try {
    let tasks = await StoreAPI.getTask();
    let doneTasks = await StoreAPI.getDoneTask(
      format(new Date(), 'dd.MM.yyyy')
    );

    if (!tasks) throw new Error('Не удалось получить данные');
    if (!doneTasks) throw new Error('Не удалось получить данные');

    dispatch(setTasks(tasks));
    dispatch(setDoneTasks(doneTasks));
    dispatch(setInitStatus(true));
  } catch (err: any) {
    return err.message as string;
  }
};
export const addTask =
  (taskName: string, time: number, repeat = 0): AppThunk =>
  async (dispatch) => {
    const taskData = { name: taskName, done: false, time, repeat } as TaskData;
    try {
      await StoreAPI.setTask(taskData);
      dispatch(getTasks());
    } catch (err: any) {
      return err.message as string;
    }
  };
export const updateTask =
  (id: string, data: PartialTaskData): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.updateTask(id, data);
      dispatch(editTask({ id, data }));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const checkTask =
  (task: Task): AppThunk =>
  async (dispatch, getState) => {
    try {
      const { id, data } = task;
      const { name, done, time, repeat } = data;
      let newData;
      // make the data repeat date or complete
      if (repeat) {
        //move task for future
        let newTime = time;
        //count expiration of task + 1 repeat
        let expiration = Math.round(
          (getTime(endOfToday()) - newTime) /
            (milliseconds({ days: 1 }) * repeat) +
            1
        );
        //count date of future task
        newTime = getTime(add(newTime, { days: repeat * expiration }));
        newData = { time: newTime };
      } else {
        newData = { done: !done };
      }
      // add task to done list
      const doneTasksList = {
        [id]: name,
      };
      //send data as batch
      const taskApiData = { taskId: id, taskData: newData };
      const doneTaskApiData = {
        doneTaskDate: format(new Date(), 'dd.MM.yyyy'),
        doneTasks: doneTasksList,
        doneTasksMerge: true,
      };
      await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);
      //update store
      dispatch(editTask({ id, data: newData }));
      dispatch(addTaskToDoneTasksList(doneTasksList));
    } catch (err: any) {
      return err.message as string;
    }
  };
export const uncheckTask =
  (id: string): AppThunk =>
  async (dispatch, getState) => {
    try {
      let task = getState().tasks.tasksList.find((task) => task.id === id);
      if (task) {
        const { done, time, repeat } = task.data;
        let newData;
        // make the data repeat date or complete
        if (repeat) {
          //move date to past
          newData = { time: getTime(add(time, { days: -repeat })) };
        } else {
          newData = { done: !done };
        }
        // add task to done list
        let doneTasksList = { ...getState().tasks.doneTasksList };
        delete doneTasksList[id];
        //send data as batch
        const taskApiData = { taskId: id, taskData: newData };
        const doneTaskApiData = {
          doneTaskDate: format(new Date(), 'dd.MM.yyyy'),
          doneTasks: doneTasksList,
          doneTasksMerge: false,
        };
        await StoreAPI.setBatchDoneTask(taskApiData, doneTaskApiData);

        //update store

        dispatch(editTask({ id, data: newData }));
        dispatch(removeTaskToDoneTasksList(id));
      } else throw new Error('Нет такой задачи');
    } catch (err: any) {
      return err.message as string;
    }
  };
export const deleteTask =
  (id: string): AppThunk =>
  async (dispatch) => {
    try {
      await StoreAPI.deleteTask(id);
      dispatch(removeTask(id));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default tasksSlice;

export interface Task {
  id: string;
  data: TaskData;
}
export interface TaskData {
  name: string;
  done: boolean;
  time: number;
  repeat: number;
}
export type PartialTaskData = Partial<TaskData>;
interface PartialTask {
  id: string;
  data: PartialTaskData;
}
export interface DoneTask {
  [id: string]: string;
}
