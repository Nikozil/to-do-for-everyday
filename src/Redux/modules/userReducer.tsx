import { User } from '@firebase/auth/dist/auth-exp-public';
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AuthAPI } from '../../api/AuthAPI';
import { UserAPI } from '../../api/UserAPI';
import { AppThunk } from '../store';

let initialState = {
  userData: { uid: '', displayName: null, email: null } as UserData,
  authStatus: false as boolean,
  initStatus: false as boolean,
  loginError: null as LoginErrorType,
};

export type initialStateType = typeof initialState;
const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    setUserData: (state, action: PayloadAction<User | UserData>) => {
      const { uid, email, displayName } = action.payload;
      return { ...state, userData: { uid, email, displayName } };
    },
    setAuthStatus: (state, action: PayloadAction<boolean>) => {
      state.authStatus = action.payload;
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
    setError: (state, action: PayloadAction<LoginErrorType>) => {
      state.loginError = action.payload;
    },
  },
});

export const { setUserData, setAuthStatus, setInitStatus, setError } =
  userSlice.actions;

export const signIn =
  (email: string, password: string, remember: boolean): AppThunk =>
  async (dispatch) => {
    try {
      await AuthAPI.signIn(email, password, remember);
      dispatch(setError(null));
    } catch (err: any) {
      dispatch(setError(err.message as string));
    }
  };
export const signOut = (): AppThunk => async (dispatch) => {
  await AuthAPI.signOut();
  dispatch(setAuthStatus(false));
  dispatch(setUserData(initialState.userData));
};
export const updateUserData = (): AppThunk => async (dispatch) => {
  await AuthAPI.updateUserStatus(
    dispatch,
    setUserData,
    setAuthStatus,
    setInitStatus
  );
};
export const updateProfile =
  (displayName: string): AppThunk =>
  async (dispatch) => {
    await UserAPI.updateProfile(displayName);
    dispatch(updateUserData());
  };

export default userSlice;

export type LoginErrorType = string | null;

export type UserData = Pick<User, 'uid' | 'displayName' | 'email'>;
