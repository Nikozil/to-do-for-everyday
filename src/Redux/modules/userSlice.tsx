import { User } from '@firebase/auth/dist/auth-exp-public';
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AuthAPI } from '../../api/AuthAPI';
import { UserAPI } from '../../api/UserAPI';
import { AppThunk } from '../store';

let initialState = {
  userData: { uid: '', displayName: null, email: null } as UserData,
  authStatus: false as boolean,
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    setUserData: (state, action: PayloadAction<User | UserData>) => {
      const { uid, email, displayName } = action.payload;
      return { ...state, userData: { uid, email, displayName } };
    },
    setAuthStatus: (state, action: PayloadAction<boolean>) => {
      state.authStatus = action.payload;
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
  },
});

export const { setUserData, setAuthStatus, setInitStatus } = userSlice.actions;

export const signIn =
  (email: string, password: string, remember: boolean): AppThunk =>
  async (dispatch) => {
    try {
      await AuthAPI.signIn(email, password, remember);
      return 'Авторизация выполнена';
    } catch (err: any) {
      return err.message as string;
    }
  };
export const signOut = (): AppThunk => async (dispatch) => {
  try {
    await AuthAPI.signOut();
    dispatch(setAuthStatus(false));
    dispatch(setUserData(initialState.userData));
  } catch (err: any) {
    return err.message as string;
  }
};
export const updateUserData = (): AppThunk => async (dispatch) => {
  await AuthAPI.updateUserStatus(
    dispatch,
    setUserData,
    setAuthStatus,
    setInitStatus
  );
};
export const updateProfile =
  (displayName: string): AppThunk<Promise<string>> =>
  async (dispatch) => {
    try {
      await UserAPI.updateProfile(displayName);
    } catch (err: any) {
      return err.message;
    }
    dispatch(updateUserData());
  };
export const updatePassword =
  (oldPassword: string, newPassword: string): AppThunk<Promise<string>> =>
  async (dispatch) => {
    try {
      let responce = await AuthAPI.updatePassword(oldPassword, newPassword);
      return responce;
    } catch (err: any) {
      return err.message as string;
    }
  };

export default userSlice;

export type UserData = Pick<User, 'uid' | 'displayName' | 'email'>;
