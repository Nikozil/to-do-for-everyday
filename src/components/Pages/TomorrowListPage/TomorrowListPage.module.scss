@import '../../../App.scss';

.tomorrow-page {
  display: flex;
  // align-items: center;
  justify-content: center;
  height: 100%;
}
.tomorrow-page__content {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  height: 95%;
  width: 100%;
  min-width: 200px;
  max-width: 600px;
  margin: 0 5px 0 5px;
  text-align: center;

  font-size: 1.5em;
  color: $primary-color;
  @media screen and (max-width: 767px) {
    font-size: 1.2rem;
    margin: 0 15px 0 10px;
  }
}
.spinner {
  margin: 0 auto 0 auto;
  color: $primary-color !important;
}

.task {
  display: flex;
  align-items: center;
}
.task__task-name {
  margin-left: 10px;
  margin-bottom: 4px;
}
.completed-tasks {
  opacity: 0.8;
}
