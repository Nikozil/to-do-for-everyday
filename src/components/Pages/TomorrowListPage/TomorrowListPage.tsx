import React from 'react';
import NewTaskWithAutocompleteForm from '../../Forms/NewTaskWithAutocompleteForm/NewTaskWithAutocompleteForm';
import PlansMap from './PlansMap/PlansMap';
import styles from './TomorrowListPage.module.scss';

const TomorrowListPage = () => {
  const duration = { days: 1 };

  return (
    <article className={styles['tomorrow-page']}>
      <div className={styles['tomorrow-page__content']}>
        <section className={styles['tomorrow-tasks']}>
          <div className={'my-4'}>
            <NewTaskWithAutocompleteForm duration={duration} />
          </div>

          <PlansMap />
        </section>
      </div>
    </article>
  );
};

export default TomorrowListPage;
