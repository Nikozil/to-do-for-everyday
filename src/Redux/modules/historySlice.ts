import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { endOfMonth, getTime, startOfMonth } from 'date-fns';

import { StoreAPI } from '../../api/StoreAPI';
import { AppThunk } from '../store';
import { LivedDay } from './tasksSlice';

export const initialState = {
  daysList: [] as HistoryDay[],
  initStatus: false as boolean,
};

export type initialStateType = typeof initialState;
const historySlice = createSlice({
  name: 'history',
  initialState,
  reducers: {
    setDays: (state, action: PayloadAction<HistoryDay[]>) => {
      return { ...state, daysList: action.payload };
    },
    setInitStatus: (state, action: PayloadAction<boolean>) => {
      state.initStatus = action.payload;
    },
  },
});

export const { setDays, setInitStatus } = historySlice.actions;

export const getDays =
  (month: number, year: number): AppThunk =>
  async (dispatch, getState) => {
    try {
      dispatch(setInitStatus(false));
      const date = getTime(new Date(year, month));
      const start = getTime(startOfMonth(date));
      const end = getTime(endOfMonth(date));

      const days = await StoreAPI.getDays(start, end);
      if (!days) throw new Error('Не удалось получить данные');

      dispatch(setDays(days));
      dispatch(setInitStatus(true));
    } catch (err: any) {
      return err.message as string;
    }
  };

export default historySlice;

export interface HistoryDay extends LivedDay {
  timestamp: number;
}
