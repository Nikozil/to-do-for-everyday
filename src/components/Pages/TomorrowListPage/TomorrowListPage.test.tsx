import React from 'react';
import { initialState, Task } from '../../../Redux/modules/tasksSlice';
import { render } from '../../../utils/tests/test-utils';
import TomorrowListPage from './TomorrowListPage';
const testTask = {
  id: '123',
  data: {
    name: 'task0',
    repeat: 0,
    done: false,
    time: new Date().getTime() + 86400000,
  },
} as Task;
describe('TomorrowListPage', () => {
  it('render correct with tasks', async () => {
    const { asFragment } = render(<TomorrowListPage />, {
      preloadedState: {
        tasks: {
          ...initialState,
          tasksList: [testTask],
          initStatus: true,
        },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });
  it('render correct without tasks', async () => {
    const { asFragment } = render(<TomorrowListPage />, {
      preloadedState: {
        tasks: {
          ...initialState,
          initStatus: true,
        },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });
  it('render correct without init', async () => {
    const { asFragment } = render(<TomorrowListPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
