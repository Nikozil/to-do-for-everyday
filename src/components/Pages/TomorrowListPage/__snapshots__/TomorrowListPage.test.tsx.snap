// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TomorrowListPage render correct with tasks 1`] = `
<DocumentFragment>
  <div
    class="tomorrow-page"
  >
    <div
      class="tomorrow-page__content"
    >
      <section
        class="tomorrow-tasks"
      >
        <div
          class="my-4"
        >
          <form
            action="#"
          >
            <div
              class="mb-1"
            >
              <input
                autocomplete="off"
                class="form-control newTask"
                id="newTask"
                name="newTask"
                placeholder="Новая Задача"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
        <h2
          class="mb-3"
        >
          Задачи на завтра
        </h2>
        <ul
          class="list"
        >
          <li
            class="content__li"
          >
            <div
              class="content content_border"
            >
              <span
                class="task"
              >
                <button
                  aria-label="delete task"
                  class="button"
                  data-testid="delete-button"
                  name="delete task"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
                    />
                  </svg>
                </button>
                <span
                  class="task__taskName"
                >
                  task0
                </span>
              </span>
              <span
                class="task__buttons"
              >
                <button
                  aria-label="repeat task"
                  class="button"
                  data-testid="repeat-button"
                  name="repeat task"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M13.828 1.686l8.486 8.486-1.415 1.414-.707-.707-4.242 4.242-.707 3.536-1.415 1.414-4.242-4.243-4.95 4.95-1.414-1.414 4.95-4.95-4.243-4.242 1.414-1.415L8.88 8.05l4.242-4.242-.707-.707 1.414-1.415zm.708 3.536l-4.671 4.67-2.822.565 6.5 6.5.564-2.822 4.671-4.67-4.242-4.243z"
                      />
                    </g>
                  </svg>
                </button>
                <button
                  aria-label="options task"
                  class="button"
                  data-testid="options-button"
                  name="options task"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14 7h0.25c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.413-0.338-0.75-0.75-0.75h-0.25v-3h-2v3h-0.25c-0.412 0-0.75 0.337-0.75 0.75v2.5c0 0.412 0.338 0.75 0.75 0.75h0.25v9h2v-9zM12 4h2v2h-2v-2zM9.25 13c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.412-0.338-0.75-0.75-0.75h-0.25v-9h-2v9h-0.25c-0.412 0-0.75 0.338-0.75 0.75v2.5c0 0.412 0.338 0.75 0.75 0.75h0.25v3h2v-3h0.25zM7 10h2v2h-2v-2zM4.25 7c0.412 0 0.75-0.338 0.75-0.75v-2.5c0-0.413-0.338-0.75-0.75-0.75h-0.25v-3h-2v3h-0.25c-0.413 0-0.75 0.337-0.75 0.75v2.5c0 0.412 0.337 0.75 0.75 0.75h0.25v9h2v-9h0.25zM2 4h2v2h-2v-2z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="task check"
                  class="button task__buttons_left-margin"
                  data-testid="check-button"
                  name="task check"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM14 14h-12v-12h12v12z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </div>
   
</DocumentFragment>
`;

exports[`TomorrowListPage render correct without init 1`] = `
<DocumentFragment>
  <div
    class="tomorrow-page"
  >
    <div
      class="tomorrow-page__content"
    >
      <section
        class="tomorrow-tasks"
      >
        <div
          class="my-4"
        >
          <form
            action="#"
          >
            <div
              class="mb-1"
            >
              <input
                autocomplete="off"
                class="form-control newTask"
                id="newTask"
                name="newTask"
                placeholder="Новая Задача"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
        <div
          class="spinner-border spinner"
          role="status"
        >
          <span
            class="visually-hidden"
          >
            Loading...
          </span>
        </div>
      </section>
    </div>
  </div>
   
</DocumentFragment>
`;

exports[`TomorrowListPage render correct without tasks 1`] = `
<DocumentFragment>
  <div
    class="tomorrow-page"
  >
    <div
      class="tomorrow-page__content"
    >
      <section
        class="tomorrow-tasks"
      >
        <div
          class="my-4"
        >
          <form
            action="#"
          >
            <div
              class="mb-1"
            >
              <input
                autocomplete="off"
                class="form-control newTask"
                id="newTask"
                name="newTask"
                placeholder="Новая Задача"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
        <h2
          class="mb-3"
        >
          Задачи на завтра
        </h2>
        <span
          class="stub"
        >
          Нет новых задач
        </span>
      </section>
    </div>
  </div>
   
</DocumentFragment>
`;
